### DISCUSSION TITLE
Opening a webhook endpoint to receive data from external server about a payment. Rowy duplicating response

### DISCUSSION THREAD

**Discussion by Ausint** [AntlerVC/firetable#1401] [Support / Q&A] [answered]
good day. 

I have for a week, been trying to setup a webhook event receiver from Yoco, a payment gateway. 

1) user on my app (built with flutterflow), checkouts and make a payment. Yoco is receiving a payment but I am trying to now verify that the payment went through and also extract the metadata from Yoco's response so that i can create an in-app order notification. 

2) I am receiving the webhook event from yoco however, it is duplicating to up until 10 results. 

I figured Yoco has the following code in their guide to setup a webhook endpoint 

// Using Express

app.post("/my/webhook/url", function(req, res) {
    const requestBody = req.rawBody;

    // Verify and process the received data

    res.send(200);
});

So i assume that res.send(200) is a response that should be sent back to the serve to notify that we have received and processed the information. I tried many things to work with that however i got the same results .

Please, if you could guide me, on what could be the suitable code I would appreciate it

**Comment by Chadha93** [comment]
Hi @Ausint, You should give a try to implement this with [BuildShip](https://buildship.com/) to handle and receive the data from an external server. 


I invite you to join our [Discord](https://www.rowy.io/discord) and post this on the community support channel. 

--
Cheers,
Gaurav Chadha | Technical Support Engineer | [Rowy](https://rowy.app/) + [BuildShip](https://buildship.com/)
backend for everyone

**Reply by Chadha93** [reply, marked as answer]
@Ausint 
To handle the webhook event and prevent duplication of responses, you can use a Cloud Function in Google Cloud Platform (GCP) written in JavaScript. Here's an example of how you can implement it:

```javascript
exports.yocoWebhook = (req, res) => {
  // Verify the request is a POST request
  if (req.method !== 'POST') {
    res.status(405).send('Method Not Allowed');
    return;
  }

  // Get the request body
  const requestBody = req.body;

  // Process the received data and extract metadata
  // Replace this with your own logic to handle the payment data
  const paymentData = requestBody.payment_data;

  // Create an in-app order notification using the metadata
  // Replace this with your own logic to create the notification

  // Send a response to acknowledge receipt of the webhook event
  res.status(200).end();
};
```

To deploy this Cloud Function, you can use the gcloud command-line tool or the GCP Console. Make sure you have the necessary permissions to deploy the function.

Once deployed, you can set up the webhook endpoint in Yoco by providing the URL of your Cloud Function. For example, if you deploy the function as `yocoWebhook`, the URL will be `https://REGION-PROJECT_ID.cloudfunctions.net/yocoWebhook`.

When Yoco sends a webhook event to this endpoint, the Cloud Function will verify the request is a POST request, process the received data, and send a response with a 200 status code to acknowledge receipt of the event. This will prevent Yoco from resending the webhook event.

Note: In the example above, I assumed you are using the Express framework. If you are not using Express, you can modify the code accordingly to handle the request and response objects in your framework of choice.
